portfolio/src/main/webapp/script.js

```diff
@@ -45,12 +45,40 @@ content.appendChild(
     link: "https://vedantroy.github.io/",
   })
 );
-fetch("/data")
-  .then((resp) => resp.json())
-  .then((commentsArray) => {
-    for (const comment of commentsArray) {
-      const p = document.createElement("p");
-      p.textContent = comment;
-      document.body.append(p);
+
+let maxComments = 3;
+const form = document.getElementById("max-comments-form");
+form.addEventListener("submit", (e) => {
+  e.preventDefault();
+  const data = new FormData(form);
+  for (const [key, val] of data) {
```

Comment:
it looks like this loop will continue iterating (well, if there were hypothetically more elements in the form) even after work is done? perhaps a more natural way of doing what you want, if i'm understanding correctly, would be to extract the key you care about, and use that directly? ie:

const maxComments = data.get("max_comments") // or default?
getComments(maxComments)